swift/Sources/PiranhaKit/CleanupStaleFlags/StaleFlagCleaner.swift

```diff
@@ -822,3 +863,33 @@ private extension TriviaPiece {
         return true
     }
 }
+
+private extension XPFlagCleaner {
+    
+    
+    final class Scope {
+        // True signifies that the replacement can happen and false signifies that replacement shouldn't happen.
+        var labelVariableMap: [String: Bool] = [:]
```

Comment:
Do you think we can update the map to not use Strings? We currently use strings (see [here](https://github.com/uber/piranha/blob/master/swift/Sources/PiranhaKit/CleanupStaleFlags/StaleFlagCleaner.swift#L42) to identify variables which is hacky. We can't identify the equivalence among self.X, X, <classname>.X and have to add specific mappings. Ideally, we will have a variable to bool mapping and the matching will be on the variable as opposed to strings.  

I am fine if you want to handle it in a followup PR too as it touches all parts of the current implementation.